src/transformStateWithClones.js

```diff
@@ -8,6 +8,30 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const newState = { ...state };
+  const arrayOfObjects = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    const oneElement = actions[i];
+
+    if (oneElement.type === 'addProperties') {
```

Comment:
you compare `oneElement.type` in all `if` statement. It's better use here `switch/case`